### PR TITLE
Get the hybrid public key only if key type is HYBRID_ENCRYPT

### PR DESCRIPTION
**Pull request by yoavamit**

### CODE REVIEW

File: misk-crypto/src/main/kotlin/misk/crypto/KmsClientModule.kt [hunk 1]
```diff
@@ -1,25 +1,72 @@
...
+    /**
+     * Do not use.
+     * If you need to initialize a [KmsClient] using the default AWS credentials lookup path please use
+     * [com.google.crypto.tink.integration.awskms.AwsKmsClient]
+     */
+    override fun withDefaultCredentials(): KmsClient {
+      throw UnsupportedOperationException("Not implemented")
+    }
+  }
 }
```
Comment (natvyas) [inline misk-crypto/src/main/kotlin/misk/crypto/KmsClientModule.kt:71]: Are all these changes needed here?

**Review by natvyas — APPROVED** [review]

File: misk-crypto/src/main/kotlin/misk/crypto/KmsClientModule.kt (hunk 1 above)
Comment (yoavamit) [reply misk-crypto/src/main/kotlin/misk/crypto/KmsClientModule.kt]: Sorry, I got a commit from another PR sneak in here :)

**Review by yan — APPROVED** [review]

**Comment by chris-ryan-square** [comment]
@yoavamit could you bring this PR up to date so it can be merged please?

File: misk-crypto/src/main/kotlin/misk/crypto/S3KeySource.kt [hunk 2]
```diff
@@ -61,10 +61,15 @@ class S3KeySource @Inject constructor(
   } ?: defaultS3
 
   // N.B. The path we're using for the object is based on _our_ region, not where the bucket lives
-  private fun objectPath(alias: String) = "$alias/${defaultS3.regionName.toLowerCase()}"
+  private fun objectPath(alias: String, type: KeyType): String {
+    if (type === KeyType.HYBRID_ENCRYPT) {
+      return "$alias/public"
+    }
+    return "$alias/${defaultS3.regionName.toLowerCase()}"
```
Comment (yan) [inline misk-crypto/src/main/kotlin/misk/crypto/S3KeySource.kt:68]: this should be `s3`, not `defaultS3`

**Pull request merged**